cosmoz-default-tree.html

```diff
@@ -23,134 +35,91 @@
-		if (propertyName === undefined || propertyValue === undefined) {
-			return;
-		}
-
-		if (propertyName === 'pathLocator') {
-			return this._getNodeByPathLocator(propertyValue);
-		}
-
-		return this._searchAllRoots(propertyName, propertyValue);
+	/**
+	 * Loops over a node list (or roots if not defined) and returns the concatenated results of search().
+	 */
+	Cosmoz.DefaultTree.prototype.searchNodes = function (propertyName, propertyValue, exact, nodes) {
+		var results = [];
+
+		// Defaults
+		exact = exact !== undefined ? exact : true;
+		nodes = nodes ? nodes : this._roots;
+
+		nodes.forEach(function (node) {
+			results = results.concat(this.search(node, propertyName, propertyValue, exact));
+		}, this);
+			
+		return results;
 	};
 
-	Cosmoz.DefaultTree.prototype.getPath = function (node) {
-		if (node === undefined) {
-			return;
+	/**
+	 * Returns the Node of a given pathLocator.
+	 */
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator) {
+		if (!pathLocator) {
+			return this._treeData;
 		}
-		return this._getPathByPathLocator(node.pathLocator);
+		var	pathNodes = this.getPathNodes(pathLocator, this._treeData);
+		return pathNodes[pathNodes.length - 1];
 	};
 
-	Cosmoz.DefaultTree.prototype.getPathByProperty = function (propertyName, propertyValue) {
-		if (propertyName === undefined || propertyValue === undefined) {
-			return;
+	/**
+	 * Returns an Array of nodes on a given path.
+	 * If pathLocator is empty or not defined, the Root objects get returned.
+	 */
+	Cosmoz.DefaultTree.prototype.getPathNodes = function (pathLocator, nodes, separatorSign) {
+		if (!pathLocator) {
+			return this._roots;
 		}
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+		// Defaults
+		separatorSign = separatorSign ? separatorSign : '.';
+
+		var path = pathLocator.split(separatorSign),
+			pathSegment = nodes ? nodes : this._treeData;
+		
+		return path.map(function (nodeKey) {
+			var node = pathSegment[nodeKey],
+				children = this.getChildren(node);
+			if (node && children !== undefined && Object.keys(children).length > 0) {
+				pathSegment = children;
+			}
+			return node;
+		}, this);
+	},
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+	/**
+	 * Returns a String (separated node comparison properties) representing the path.
+	 */
+	Cosmoz.DefaultTree.prototype.getPathString = function (pathLocator, comparisonProperty, pathSeparator) {
+		var nodesOnPath = this.getPathNodes(pathLocator, this._treeData);
+
+		// Defaults
+		pathSeparator = pathSeparator ? pathSeparator : '/';
+
+		return nodesOnPath.map(function (node) {
+			return node[comparisonProperty];
+		}).join(pathSeparator);
 	};
 
+	/**
+	 * Returns an Object or an Array representing the children of a node.
+	 */
 	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
```

Comment:
Having a function that only returns the `children` property of the argument seems pretty pointless, then it should be reversed to how it was before this PR - `return _objectValues(node.children);` as I commented on the last review.
For any cases where we _don't_ need to convert it to an array, we can just use `node.children` instead of this method.